src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -201,11 +202,22 @@ private static bool IsTupleInDeconstruction(SyntaxNode tuple)
                 {
                     return true;
                 }
-            }
-            while ((tuple = parent.Parent).IsKind(SyntaxKind.TupleExpression)
-                && (parent = parent.Parent.Parent) != null);
 
-            return false;
+                tuple = parent.Parent;
+                if (tuple.IsKind(SyntaxKind.TupleExpression))
+                {
+                    parent = parent.Parent.Parent;
+                    if (parent == null)
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            while (true);
```

Comment:
consdier making this "while (parent != null)".   Consider moving the ```return false``` out of hte do-while.